src/map/pc.cpp

```diff
@@ -5494,21 +5494,21 @@ enum e_setpos pc_setpos(struct map_session_data* sd, unsigned short mapindex, in
 	if( sd->state.changemap ) { // Misc map-changing settings
 		int i;
 
-		if(map_getmapdata(sd->bl.m)->instance_id && !mapdata->instance_id) {
-			bool instance_found = false;
+		if (map_getmapdata(sd->bl.m)->instance_id && map_getmapdata(sd->bl.m)->instance_id != mapdata->instance_id) {
```

Comment:
You could store this value like: `if ((instance_id = map_getmapdata(sd->bl.m)->instance_id) && instance_id != mapdata->instance_id)`.

Then `instance_id` could be used in the checks below as well.